File: Using python/Measures of Central Tendency and Dispersion/1/1.py
data = {'ID':[1,2,3,4,5,6,7,8,9,10],'Name':['Ram','Shyam','Hari','Sita','Anand','Sunny','Sweta','Dharmesh','Dipanjan','Sinu'],'Gender':['M','M','M','F','M','M','F','M','M','F'],'DO_Birth':["01/01/1998", "09/23/1998", "11/15/1998", "05/11/1998","02/27/1998","06/01/1998","06/14/1998","03/31/1998","07/08/1998","11/30/1998"],'Do_Admission':["01/01/2001", "09/23/2001", "11/15/2005", "05/11/2001","02/27/2002","06/01/2008","06/14/2006","03/31/2005","07/08/2006","11/30/2013"],'Payment_Made':['Y','N','N','Y','Y','Y','N','Y','N','Y'],'Mp_Percentage':[85,82,80,72,70,92,94,89,86,69],'Hs_Percentage':[89,82,72,80,92,94,86,69,70,79]}
studata = pd.DataFrame(data)
studata['DO_Birth']=pd.to_datetime(studata['DO_Birth'])
studata['Do_Admission']=pd.to_datetime(studata['Do_Admission'])


import statistics as stat
stat.mean(studata['Mp_Percentage'])
stat.median(studata['Mp_Percentage'])



stat.mean(studata[studata['Gender']=='M']['Mp_Percentage'])


stat.mean(studata[studata['Gender']=='F']['Mp_Percentage'])





import pandas as pd
data = {'ID':[1,2,3,4,5,6,7,8,9,10],'Name':['Ram','Shyam','Hari','Sita','Anand','Sunny','Sweta','Dharmesh','Dipanjan','Sinu'],'Gender':['M','M','M','F','M','M','F','M','M','F'],'DO_Birth':["01/01/1998", "09/23/1998", "11/15/1998", "05/11/1998","02/27/1998","06/01/1998","06/14/1998","03/31/1998","07/08/1998","11/30/1998"],'Do_Admission':["01/01/2001", "09/23/2001", "11/15/2005", "05/11/2001","02/27/2002","06/01/2008","06/14/2006","03/31/2005","07/08/2006","11/30/2013"],'Payment_Made':['Y','N','N','Y','Y','Y','N','Y','N','Y'],'Mp_Percentage':[85,82,80,72,70,92,94,89,86,69],'Hs_Percentage':[89,82,72,80,92,94,86,69,70,79]}
studata = pd.DataFrame(data)
studata['DO_Birth']=pd.to_datetime(studata['DO_Birth'])
studata['Do_Admission']=pd.to_datetime(studata['Do_Admission'])

import statistics as stat
from datetime import date
ans=[]
test=studata['DO_Birth']
for i in range(9):
    x=(pd.to_datetime("2021-04-25")-pd.to_datetime(test[i])).days
    ans.append(x)
print(stat.mean(ans))


stat.mean(ans)/365

round(stat.mean(ans)/365)


import pandas as pd
data = {'ID':[1,2,3,4,5,6,7,8,9,10],'Name':['Ram','Shyam','Hari','Sita','Anand','Sunny','Sweta','Dharmesh','Dipanjan','Sinu'],'Gender':['M','M','M','F','M','M','F','M','M','F'],'DO_Birth':["01/01/1998", "09/23/1998", "11/15/1998", "05/11/1998","02/27/1998","06/01/1998","06/14/1998","03/31/1998","07/08/1998","11/30/1998"],'Do_Admission':["01/01/2001", "09/23/2001", "11/15/2005", "05/11/2001","02/27/2002","06/01/2008","06/14/2006","03/31/2005","07/08/2006","11/30/2013"],'Payment_Made':['Y','N','N','Y','Y','Y','N','Y','N','Y'],'Mp_Percentage':[85,82,80,72,70,92,94,89,86,69],'Hs_Percentage':[89,82,72,80,92,94,86,69,70,79]}
studata = pd.DataFrame(data)
studata['DO_Birth']=pd.to_datetime(studata['DO_Birth'])
studata['Do_Admission']=pd.to_datetime(studata['Do_Admission'])

import statistics as stat
from datetime import date
ans=[]
test=studata['DO_Birth']
test1=studata['Gender']
for i in range(9):
    if(test1[i]=='M'):
        x=(pd.to_datetime("2021-04-25")-pd.to_datetime(test[i])).days
        ans.append(x)
print(stat.mean(ans))


import numpy as np
print(studata.describe())
print(Campus_Employement.describe())
a=len(studata[studata['Gender']=='M'].axes[0])
b=len(studata[studata['Gender']=='F'].axes[0])
print(a,":",b)

File: Using R/Measures of Central Tendency and Dispersion/1/1.R
id <- c(1:10)
name <- c('Ram','Shyam','Hari','Sita','Anand','Sunny','Sweta','Dharmesh','Dipanjan','Sinu')
gender <- c('M','M','M','F','M','M','F','M','M','F')
do_birth <- as.Date(c("1998-01-01", "1998-09-23", "1998-11-15", "1998-05-11","1998-02-27","1998-06-01","1998-06-14","1998-03-31","1998-07-08","1998-11-30"))
do_adm <- as.Date(c("2003-01-01", "2004-09-23", "2005-11-15", "2003-05-11","2007-02-27","2008-06-01","2006-06-14","2005-03-31","2006-07-08","2013-11-30"))
payment_made <- c('Y','N','N','Y','Y','Y','N','Y','N','Y')
mp_perc <- c(85,82,80,72,70,92,94,89,86,69)
hs_perc <- c(89,82,72,80,92,94,86,69,70,79)

studata <- data.frame(ID=id,Name=name,Gender=gender,DO_Birth=do_birth,
DO_Admission=do_adm,Payment_Made=payment_made,MP_Percentage=mp_perc,HS_Percentage=hs_perc)


year <- c(2016,2016,2017,2017,2018,2018,2019,2019,2020,2020)
stream=c("BCA","CSE","BCA","CSE","BCA","CSE","BCA","CSE","BCA","CSE")
tot_count =c(16,20,25,30,35,50,40,59,80,100)
Total_Employement_Placement =data.frame(Year=year,Stream=stream,Total_Count=tot_count)


library(DescTools)
library(modeest)

mean(studata$MP_Percentage)

median(studata$MP_Percentage)

mean(studata[studata$Gender=="M",]$MP_Percentage)
mean(studata[studata$Gender=="F",]$MP_Percentage)

summary(studata)
summary(Total_Employement_Placement)
mean(as.numeric(Sys.Date()-studata$DO_Birth))
mean(as.numeric(Sys.Date()-studata[studata$Gender=="M",]$DO_Birth))


a <- mean(as.numeric(Sys.Date()-studata[studata$Gender=="M",]$DO_Birth))

a <- mean(as.numeric(Sys.Date()-studata[studata$Stream=="BCA",]$DO_Birth))
year = a/365
round(year)

a <- nrow(studata[studata$Gender == "M",])
b <- nrow(studata[studata$Gender == "F",])

cat(a,":",b,"\n")